### ISSUE TITLE
RDBBackfill unit test failure

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#4370]
Got this when running the test suite on `raft` https://github.com/rethinkdb/rethinkdb/commit/d14f50c585f040b899c2373a065188a845100765 in one of the RDBBackfill unit tests:

```
Version: rethinkdb 2.0.0-1-1124-gd14f50 (debug) (CLANG 3.2 (tags/RELEASE_32/final))
error: Error in ./src/unittest/clustering_utils.hpp at line 179:
Previous message may not have been written to the log file (cannot open or find log file).
error: Guarantee failed: [it->second == response] For key `jqmoonvvquopecmuiyda`: expected ``, got `9796`
Previous message may not have been written to the log file (cannot open or find log file).
error: Backtrace:
Previous message may not have been written to the log file (cannot open or find log file).
error: Tue Jun  9 18:35:19 2015
       
       1: rethinkdb_backtrace(void**, int) at rethinkdb_backtrace.cc:100
       2: backtrace_t::backtrace_t() at backtrace.cc:203
       3: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at backtrace.cc:283
       4: format_backtrace(bool) at backtrace.cc:198
       5: report_fatal_error(char const*, int, char const*, ...) at errors.cc:83
       6: unittest::test_inserter_t::validate() at clustering_utils.hpp:178
       7: unittest::dispatcher_inserter_t::validate() at rdb_backfill.cc:78
       8: unittest::run_backfill_test(unittest::backfill_test_config_t const&) at rdb_backfill.cc:331
       9: unittest::run_RDBBackfill_LargeTable() at rdb_backfill.cc:392
       10: std::_Function_handler<void (), void (*)()>::_M_invoke(std::_Any_data const&) at functional:1779
       11: std::function<void ()>::operator()() const at functional:2160
       12: starter_t::run_wrapper(std::function<void ()> const&) at runtime.cc:61
       13: std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)>::operator()(starter_t*, std::function<void ()> const&) const at functional:550
       14: _ZNSt5_BindIFSt7_Mem_fnIM9starter_tFvRKSt8functionIFvvEEEEPS1_S4_EE6__callIvJEJLi0ELi1EEEET_OSt5tupleIJDpT0_EESt12_Index_tupleIJXspT1_EEE+0x70 at functional:1145
       15: _ZNSt5_BindIFSt7_Mem_fnIM9starter_tFvRKSt8functionIFvvEEEEPS1_S4_EEclIJEvEET0_DpOT_+0x26 at functional:1203
       16: std::_Function_handler<void (), std::_Bind<std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)> (starter_t*, std::function<void ()>)> >::_M_invoke(std::_Any_data const&) at functional:1779
       17: std::function<void ()>::operator()() const at functional:2160
       18: callable_action_instance_t<std::function<void ()>&>::run_action() at callable_action.hpp:31
       19: callable_action_wrapper_t::run() at runtime_utils.cc:43
       20: coro_t::run() at coroutines.cc:205
       21: coro_t* coro_t::spawn_sometime<std::function<void ()>&>(std::function<void ()>&&&) at coroutines.hpp:59
       22: starter_t::on_thread_switch() at runtime.cc:57
       23: linux_message_hub_t::on_event(int) at message_hub.cc:154
       24: epoll_event_queue_t::run() at epoll.cc:115
       25: linux_thread_pool_t::start_thread(void*) at thread_pool.cc:161
       26: /lib/x86_64-linux-gnu/libpthread.so.0(+0x7e9a) [0x7fd46c2a3e9a] at 0x7fd46c2a3e9a (/lib/x86_64-linux-gnu/libpthread.so.0)
       27: clone+0x6d at 0x7fd46bfd08bd (/lib/x86_64-linux-gnu/libc.so.6)
Previous message may not have been written to the log file (cannot open or find log file).
error: Exiting.
Previous message may not have been written to the log file (cannot open or find log file).
Command '['/home/ssd3/daniel/rethinkdb/build/debug_clang/rethinkdb-unittest', '-
```

Unfortunately I cannot reproduce this at the moment.

**Comment by danielmewes** [comment]
This happened during the RDBBackfill.LargeTable test. Here's the standard output from the test run:

```
Note: Google Test filter = RDBBackfill.*
[==========] Running 12 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 12 tests from RDBBackfill
[ RUN      ] RDBBackfill.DenseChanges
[       OK ] RDBBackfill.DenseChanges (18354 ms)
[ RUN      ] RDBBackfill.SparseChanges
[       OK ] RDBBackfill.SparseChanges (9496 ms)
[ RUN      ] RDBBackfill.LargeValues
[       OK ] RDBBackfill.LargeValues (11290 ms)
[ RUN      ] RDBBackfill.VeryLargeValues
[       OK ] RDBBackfill.VeryLargeValues (61344 ms)
[ RUN      ] RDBBackfill.SmallValues
[       OK ] RDBBackfill.SmallValues (3874 ms)
[ RUN      ] RDBBackfill.LargeTable
Traceback (most recent call last):
  File "/home/ssd3/daniel/rethinkdb/test/common/test_framework.py", line 594, in run
    self.test.run()
  File "/home/ssd3/daniel/rethinkdb/test/common/unit.py", line 51, in run
    subprocess.check_call([self.unit_executable, "--gtest_filter=" + filter])
  File "/usr/lib/python2.7/subprocess.py", line 511, in check_call
    raise CalledProcessError(retcode, cmd)
CalledProcessError: Command '['/home/ssd3/daniel/rethinkdb/build/debug_clang/rethinkdb-unittest', '--gtest_filter=RDBBackfill.*']' returned non-zero exit status -5
```

**Comment by danielmewes** [comment]
I can reproduce this about one out of 10 times:
`./run -j 10 -r 10 unit.RDBBackfill`

Looking into it.

**Comment by timmaxw** [comment]
In CR 3018.

**Comment by timmaxw** [comment]
Fix merged in ac050a4fe63018fc0cad02f22823aba313522c88.
